src/libraries/System.Net.Security/src/System/Net/Security/Pal.OSX/SafeDeleteSslContext.cs

```diff
@@ -14,11 +14,12 @@ namespace System.Net
 {
     internal sealed class SafeDeleteSslContext : SafeDeleteContext
     {
+        private const int InitialBufferSize = 4096 * 4;
```

Comment:
This seems pretty large. In the context of "write small buffer, wait, read small buffer, repeat" of making HTTP/1.1 calls with tiny content, will most of this buffer space be unused?

Reply:
I updated the initial size to 2K to fit a handshake with a certificate. It does not seem to impact overall performance much but allocates less.

|                          Method |         Mean |        Error |       StdDev |       Median |          Min |          Max |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|-------------------------------- |-------------:|-------------:|-------------:|-------------:|-------------:|-------------:|-------:|------:|------:|----------:|
|                  HandshakeAsync | 84,778.85 us | 1,644.491 us | 1,688.772 us | 84,146.37 us | 82,679.20 us | 88,079.59 us |      - |     - |     - |   60220 B |
| TLS12HandshakeECDSA256CertAsync | 17,574.54 us | 1,802.173 us | 2,075.385 us | 16,644.06 us | 15,521.38 us | 21,994.76 us |      - |     - |     - |   50448 B |
|  TLS12HandshakeRSA1024CertAsync |  9,503.96 us |   281.059 us |   312.397 us |  9,359.51 us |  9,214.32 us | 10,255.90 us |      - |     - |     - |   49225 B |
|  TLS12HandshakeRSA2048CertAsync | 20,251.44 us | 1,120.917 us | 1,290.850 us | 19,691.10 us | 19,073.51 us | 23,460.19 us |      - |     - |     - |   50940 B |
|  TLS12HandshakeRSA4096CertAsync | 85,768.58 us | 1,632.837 us | 1,447.468 us | 85,312.39 us | 83,928.80 us | 89,295.98 us |      - |     - |     - |   53976 B |
|                  WriteReadAsync |     21.10 us |     0.569 us |     0.633 us |     20.90 us |     20.52 us |     22.42 us | 0.1000 |     - |     - |     352 B |
|                  ReadWriteAsync |     23.96 us |     0.365 us |     0.324 us |     23.85 us |     23.57 us |     24.79 us | 0.1000 |     - |     - |     361 B |
|             ConcurrentReadWrite |     28.93 us |     0.524 us |     0.465 us |     29.03 us |     27.67 us |     29.55 us | 0.1600 |     - |     - |     513 B |
|  ConcurrentReadWriteLargeBuffer |     34.52 us |     1.483 us |     1.708 us |     34.66 us |     32.12 us |     37.65 us | 0.1000 |     - |     - |     522 B |